siad/daemon.go

```diff
@@ -66,7 +71,7 @@ func startDaemon() error {
 	if err != nil {
 		return err
 	}
-	srv, err := api.NewServer(config.Siad.APIaddr, state, gateway, host, hostdb, miner, renter, tpool, wallet, nil)
+	srv, err := api.NewServer(config.Siad.APIaddr, state, gateway, host, hostdb, miner, miningpool, renter, tpool, wallet, nil)
```

Comment:
if we're gonna stick `nil` checks everywhere anyway, we might as well keep it all in one binary.